crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -132,45 +146,178 @@ impl MarkdownPreviewView {
+                    cx.observe(workspace, |this, workspace, cx| {
+                        let item = workspace.read(cx).active_item(cx);
+                        this.workspace_updated(item, cx);
+                    })
+                    .detach();
+                } else {
+                    log::error!("Failed to listen to workspace updates");
+                }
             }
+
+            this
         })
     }
 
+    fn workspace_updated(
+        &mut self,
+        active_item: Option<Box<dyn ItemHandle>>,
+        cx: &mut ViewContext<Self>,
+    ) {
+        if let Some(item) = active_item {
+            if item.item_id() != cx.entity_id() {
+                if let Some(editor) = item.act_as::<Editor>(cx) {
+                    if Self::is_markdown_file(&editor, cx) {
+                        self.set_editor(editor, cx);
+                    }
+                }
+            }
+        }
+    }
+
+    fn is_markdown_file<V>(editor: &View<Editor>, cx: &mut ViewContext<V>) -> bool {
+        let language = editor.read(cx).buffer().read(cx).language_at(0, cx);
+        language
+            .map(|l| l.name().as_ref() == "Markdown")
+            .unwrap_or(false)
+    }
+
+    fn set_editor(&mut self, editor: View<Editor>, cx: &mut ViewContext<Self>) {
+        if let Some(active) = &self.active_editor {
+            if active.editor == editor {
+                return;
+            }
+        }
+
+        let subscription = cx.subscribe(&editor, |this, editor, event: &EditorEvent, cx| {
+            match event {
+                EditorEvent::Edited => {
+                    this.on_editor_edited(cx);
+                }
+                EditorEvent::SelectionsChanged { .. } => {
+                    let editor = editor.read(cx);
+                    let selection_range = editor.selections.last::<usize>(cx).range();
+                    this.selected_block = this.get_block_index_under_cursor(selection_range);
+                    this.list_state.scroll_to_reveal_item(this.selected_block);
+                    cx.notify();
+                }
+                _ => {}
+            };
+        });
+
+        self.tab_description = editor
+            .read(cx)
+            .tab_description(0, cx)
+            .map(|tab_description| format!("Preview {}", tab_description));
+
+        self.active_editor = Some(EditorState {
+            editor,
+            _subscription: subscription,
+        });
+
+        self.on_editor_edited(cx);
+    }
+
+    fn on_editor_edited(&mut self, cx: &mut ViewContext<Self>) {
+        if let Some(state) = &self.active_editor {
+            let editor = state.editor.read(cx);
+            let contents = editor.buffer().read(cx).snapshot(cx).text();
+            let file_location = MarkdownPreviewView::get_folder_for_active_editor(editor, cx);
+            let language_registry = self.language_registry.clone();
+            cx.spawn(move |view, mut cx| async move {
```

Comment:
I'm a bit concerned that excessive typing might cause bad things here if the markdown is large enough: we seem to reparse it on each edit (every new letter/paste in the editor _per caret_), we never cancel old reparses and all of this is done on the main thread.

We should another `Option` task field into `&mut self` and use `replace` here, ideally with some debouncing.
Also, we'd want to do `cx.background_executor().spawn` around the `parse_markdown`/